support-frontend/assets/components/paymentFrequencyTabs/paymentFrequencyTabsContainer.tsx

```diff
@@ -50,15 +60,23 @@ export function PaymentFrequencyTabsContainer({
 		dispatch(setPaymentMethod(paymentMethodToSelect));
 	}
 
-	const tabs: TabProps[] = contributionTypes[countryGroupId].map(
-		({ contributionType }) => {
-			return {
-				id: contributionType,
-				labelText: toHumanReadableContributionType(contributionType),
-				selected: contributionType === productType,
-			};
-		},
-	);
+	const isTab = (tab: TabProps | undefined): tab is TabProps => {
+		return !!tab;
+	};
+
+	const tabs: TabProps[] = contributionTypes[countryGroupId]
+		.map(({ contributionType }) => {
+			if (hideOneOff && contributionType === 'ONE_OFF') {
+				return;
+			} else {
+				return {
+					id: contributionType,
+					labelText: toHumanReadableContributionType(contributionType),
+					selected: contributionType === productType,
+				};
+			}
+		})
+		.filter(isTab) as TabProps[];
```

Comment:
Just curious- what's the TS error here if you remove `as TabProps[]`? It's better to avoid using type assertions if we can.

Reply:
Will try to remove as many as possible going forwards. Removed the need for type assertion by changing the return type of line72 function 'toHumanReadableContributionType' to string (from 'Single' | 'Monthly' | 'Annual' this did'nt associate with a string)

Reply:
Change applied

Reply:
I'm not sure that that's the correct solution- there's no type conflict at the moment between using a function that returns `'Single' | 'Monthly' | 'Annual'` to create the value of a field of type `string`, and generally it's better to avoid making types _less_ specific.

You may get a more informative error message if you move the type guard function inline, like this:

```typescript
.filter((maybeTab): maybeTab is TabProps => !!maybeTab);
```

Reply:
Placing the filter first in the EDIT avoided the type conflict in the error guard, have implemented like above and amended the function back to be more type specific ('Single' | 'Monthly' | 'Annual') again. Thanks.